client/packages/common/src/ui/layout/tables/material-react-table/useBaseMaterialTable.tsx

```diff
@@ -1,22 +1,21 @@
-import { useMemo, useRef, useState } from 'react';
+import { useMemo } from 'react';
 import {
-  getDefaultColumnOrderIds,
   MRT_RowData,
-  MRT_StatefulTableOptions,
   MRT_TableOptions,
   useMaterialReactTable,
 } from 'material-react-table';
-import {
-  getSavedTableState,
-  useTableLocalStorage,
-} from './useTableLocalStorage';
 import { useIntlUtils, useTranslation } from '@common/intl';
 import { ColumnDef } from './types';
 import { useMaterialTableColumns } from './useMaterialTableColumns';
 import { getGroupedRows } from './utils';
 import { useTableFiltering } from './useTableFiltering';
-import { getTableDisplayOptions } from './getTableDisplayOptions';
+import { useTableDisplayOptions } from './useTableDisplayOptions';
 import { useUrlSortManagement } from './useUrlSortManagement';
+import { useColumnDensity } from './tableState/useColumnDensity';
+import { useColumnOrder } from './tableState/useColumnOrder';
+import { useColumnSizing } from './tableState/useColumnSizing';
+import { useColumnVisibility } from './tableState/useColumnVisibility';
+import { useColumnPinning } from './tableState/useColumnPinning';
```

Comment:
Oh yeah, one other little thing -- you want to add an `index.ts` to the tableState folder so these can all be one `import`?